### DISCUSSION TITLE
Python file with a pickle file (of Machine Learning model pipeline)

### DISCUSSION THREAD

**Discussion by Nee-DS** [pyinstaller/pyinstaller#7686] [Help]
Hi,
My aim is to get an exe file out of the Python file which has a predict function for Bertopic ML model, and so has a pickle file. I was wondering if anyone has created an exe file from a Python file having a pickle file of the ML model. I am not able to convert that file by pyinstaller.
Any help will be appreciated.

Thanks

**Comment by rokm** [comment]
One pitfall of loading pickled objects in PyInstaller-frozen application is that the python modules required to (re)instantiate those objects are not seen by our import analysis. So you might need to figure out what modules your ML model is using, and specify them as hidden imports.

**Reply by Nee-DS** [reply]
[12948] WARNING: file already exists but should not: C:\Users\3592157\AppData\Local\Temp\1\_MEI129482\torch\_C.cp39-win_amd64.pyd
[12948] WARNING: file already exists but should not: C:\Users\3592157\AppData\Local\Temp\1\_MEI129482\torch\_C_flatbuffer.cp39-win_amd64.pyd
Traceback (most recent call last):
  File "Predict_BERT.py", line 31, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\3592157\\AppData\\Local\\Temp\\1\\_MEI129482\\W_X_Bert_domain_words.pkl'
[31656] Failed to execute script 'Predict_BERT' due to unhandled exception!

The pickle file is already in the folder, I don't know why it is still searching. Any suggestions?

**Reply by rokm** [reply]
> `FileNotFoundError: [Errno 2] No such file or directory: 'C:\Users\3592157\AppData\Local\Temp\1\_MEI129482\W_X_Bert_domain_words.pkl'`

This looks like exactly what the error message says - you are not adding the pickle file to your onefile build, so it is not found in the run-time temporary directory where the application is unpacked.

**Comment by MeheRRVardhan** [comment]
Is there anyway to do this!!
I am trying to add the pickle file and python file into executable <<PYINSTALLER>>

(myvenv) PS C:\Users\vikram\OneDrive\Desktop\LOGGER\dist> .\emotional.exe
Traceback (most recent call last):
  File "emotional.py", line 4, in <module>
  File "joblib\numpy_pickle.py", line 648, in load
  File "joblib\numpy_pickle.py", line 577, in _unpickle
  File "pickle.py", line 1205, in load
  File "pickle.py", line 1530, in load_stack_global
  File "pickle.py", line 1572, in find_class
ModuleNotFoundError: No module named 'sklearn'
[PYI-12448:ERROR] Failed to execute script 'emotional' due to unhandled exception!

**Reply by rokm** [reply]
If you want to load pickled files in the frozen application, you will need to manually specify all modules used by the pickled file as [hidden imports](https://pyinstaller.org/en/stable/usage.html#cmdoption-hidden-import). Because those modules are not seen by PyInstaller's build-time import analysis, and will therefore not be included in the frozen application.

**Reply by MeheRRVardhan** [reply]
I will give a try, Thanks @rokm, got it thank you so MUCH
